ddp-workspace/projects/ddp-dsm-ui/src/app/stool-upload/stool-upload.component.ts

```diff
@@ -0,0 +1,280 @@
+    this.route.queryParams.subscribe(params => {
+      const realm = params[ DSMService.REALM ] || null;
+      if (realm != null && realm !== '') {
+        //        this.compService.realmMenu = realm;
+        this.checkRight();
+      }
+    });
+  }
+
+  private checkRight(): void {
+    this.allowedToSeeInformation = false;
+    this.additionalMessage = null;
+    let jsonData: any[];
+    this.dsmService.getRealmsAllowed(Statics.SHIPPING).subscribe({
+      next: data => {
+        jsonData = data;
+        jsonData.forEach((val) => {
+          if (localStorage.getItem(ComponentService.MENU_SELECTED_REALM) === val) {
+            this.allowedToSeeInformation = true;
+            this.getPossibleKitType();
+          }
+        });
+        if (!this.allowedToSeeInformation) {
+          this.additionalMessage = 'You are not allowed to see information of the selected study at that category';
+        }
+      },
+      error: () => null
+    });
+  }
+
+  ngOnInit(): void {
+    if (localStorage.getItem(ComponentService.MENU_SELECTED_REALM) != null) {
+      this.checkRight();
+    } else {
+      this.additionalMessage = 'Please select a study';
+    }
+    window.scrollTo(0, 0);
+  }
+
+  getPossibleKitType(): void {
+    if (this.isSelectedRealm()) {
+      this.loading = true;
+      let jsonData: any[];
+      this.dsmService.getKitTypes(localStorage.getItem(ComponentService.MENU_SELECTED_REALM)).subscribe({
+        next: data => {
+          this.kitTypes = [];
+          jsonData = data;
+          jsonData.forEach((val) => {
+            const kitType = KitType.parse(val);
+            if(kitType.kitId===15){
+              this.kitTypes.push(kitType);
+            }
+          });
+          if (this.kitTypes.length === 1) {
+            this.kitType = this.kitTypes[ 0 ];
+          } else {
+            this.errorMessage = 'Study does not have stool kit';
+          }
+          // console.info(`${this.kitTypes.length} kit types received: ${JSON.stringify(data, null, 2)}`);
+          this.loading = false;
+        },
+        error: err => {
+          if (err._body === Auth.AUTHENTICATION_ERROR) {
+            this.auth.logout();
+          }
+          this.loading = false;
+          this.errorMessage = 'Error - Loading kit types\n' + err;
+        }
+      });
+    }
+  }
+
+  upload(): void {
+    this.errorMessage = null;
+    this.additionalMessage = null;
+    this.failedParticipants = [];
+    this.loading = true;
+    this.dsmService.uploadStoolTxtFile(
+      localStorage.getItem(ComponentService.MENU_SELECTED_REALM), this.kitType.name,
+      this.file, this.skipAddressValidation
+    )
+      .subscribe({
+        next: data => {
+          this.loading = false;
+          if (typeof data === 'string') {
+            this.errorMessage = 'Error - Uploading txt.\n' + data;
+          } else {
+
+            const result = Result.parse(data);
```

Comment:
if you change backend PR according to the comments you will have error here because of `Result` object.